src/common/transformations/src/transformations/common_optimizations/matmul_multiply_fusion.cpp

```diff
@@ -12,13 +12,20 @@
 #include "openvino/op/reshape.hpp"
 #include "openvino/op/transpose.hpp"
 #include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/rt_info/dequantization_node.hpp"
 #include "transformations/utils/utils.hpp"
 
 using namespace ov;
 
 static std::shared_ptr<Node> fuse_const_to_weights(const std::shared_ptr<Node>& matmul,
                                                    const Output<Node>& weights,
                                                    std::shared_ptr<ov::op::v0::Constant> mul_const) {
+    // skip in case the transformation can affect nncf/lpt pipelines to insert FakeQuantize
+    if (weights.get_element_type().is_quantized() || is_dequantization_node(weights.get_node_shared_ptr()) ||
+        !ov::op::util::is_on_constant_path(weights)) {
```

Comment:
`ov::op::util::is_on_constant_path` execution might be very expensive so we need to check how this change affects model compilation/conversion time

Reply:
Yes, I strongly agree. Have tried to find any alternative but did not find smt good and not corrupted. Do you have some advice?

Reply:
@v-Golubev In general, looks ok. My suggestion is to enable this in separated PR in case if performance is ok. In other case I will add this functionality in this PR

Reply:
@v-Golubev @daniil-lyakhov So sorry for long PR! I worked on other activities..

@v-Golubev I had performance degradation over the PR with using `ov::op::util::is_on_constant_path`. So, decided to implement approach based on your suggestion: to mark const nodes.

Please review. I will ask to run performance validation over my PR.
Thanks!